### PR TITLE
[PhpUnitBridge] Add some more native types

### PR DESCRIPTION
**Pull request by ghost**
| Q             | A
| ------------- | ---
| Branch?       | 7.0 
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | ~ 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/ClockMock.php [hunk 1]
```diff
@@ -71,7 +63,7 @@ public static function usleep($us)
         }
     }
 
-    public static function microtime($asFloat = false)
+    public static function microtime($asFloat = false): string
```
Comment (stof) [inline src/Symfony/Bridge/PhpUnit/ClockMock.php]: this is a wrong return type

File: src/Symfony/Bridge/PhpUnit/ClockMock.php [hunk 2]
```diff
@@ -108,10 +96,8 @@ public static function gmdate($format, $timestamp = null)
         return \gmdate($format, $timestamp);
     }
 
-    /**
-     * @return array|int|float
-     */
-    public static function hrtime($asNumber = false)
+
+    public static function hrtime($asNumber = false): array|int|float
```
Comment (stof) [inline src/Symfony/Bridge/PhpUnit/ClockMock.php]: The min PHP version of the phpunit-bridge is PHP 7.2 (this bridge has a special status regarding the min PHP version as we use it to test older maintained branches). So the union type cannot become a native return type.

File: src/Symfony/Bridge/Twig/Mime/NotificationEmail.php [hunk 3]
```diff
@@ -82,9 +80,7 @@ public function markAsPublic(): static
         return $this;
     }
 
-    /**
-     * @return $this
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Mime/NotificationEmail.php]: this must be kept as it is more specific than `static`. This is true for all methods using it.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/PhpUnit/ClockMock.php (hunk 1 above)
Comment (ghost) [reply src/Symfony/Bridge/PhpUnit/ClockMock.php]: I understand, but what type of float or mixed is recommended?

Comment (stof) [reply src/Symfony/Bridge/PhpUnit/ClockMock.php]: this should use `@return string|float`, as done for the signature of `microtime`. But this must be done in phpdoc due to the union type.

**Review by nicolas-grekas — APPROVED** [review]
I've force-pushed some more native types and reverted the union ones since they're not compatible with PHP 7.2

**Comment by nicolas-grekas** [comment]
Thank you @d-eff-it.

**Pull request merged**